synapse/third_party_rules/access_rules.py

```diff
@@ -176,12 +173,12 @@ def on_create_room(
 
             access_rule = default_rule
 
-        # Check that the preset or the join rule in use is compatible with the access
-        # rule, whether it's a user-defined one or the default one (i.e. if it involves
-        # a "public" join rule, the access rule must be "restricted").
+        # Check that the preset in use is compatible with the access rule, whether it's
+        # user-defined or the default.
         if (
-            join_rule == JoinRules.PUBLIC or preset == RoomCreationPreset.PUBLIC_CHAT
-        ) and access_rule != AccessRules.RESTRICTED:
+            preset == RoomCreationPreset.PUBLIC_CHAT
+            and access_rule != AccessRules.RESTRICTED
+        ):
```

Comment:
We should keep a check on the join rules; otherwise it is trivial to create an unrestricted room that everyone can join as long as they have the room ID (which can be circulated oob).

Reply:
I don't understand. I don't see a different with the case of a room that we create unrestricted, and for which we change later the join_rule to be public.

I suggested removing this check to support this config directly at the room creation.

Reply:
This conversation has been continued and resolved at https://github.com/matrix-org/synapse-dinsic/pull/63#issuecomment-691228423.